### PR TITLE
[Breaking Change]: Replace babel-istanbul-plugin and replace it by a webpack loader

### PR DESCRIPTION
**Pull request by yannbf**
Closes https://github.com/storybookjs/addon-coverage/issues/29
Closes https://github.com/storybookjs/addon-coverage/issues/17

# What Changed

The `babel-istanbul-plugin` was replaced by a custom webpack loader to enable support for all Webpack5 projects, independently, whether they are using Babel as a compiler or another solution, like SWC

## Release Notes

- [BREAKING CHANGE]: Dropping Storybook < 7 support
- Replaced `babel-istanbul-plugin` with a custom Webpack loader to support coverage independently of the compiler used for Webpack5 projects. Before, the coverage addon only worked if Babel was used with Webpack5. Now, users might use other compilers like, for example, SWC.

<details>
  <summary>📦 Published PR as canary version: <code>1.0.0-canary.27.12a4851.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-coverage@1.0.0-canary.27.12a4851.0
  # or 
  yarn add @storybook/addon-coverage@1.0.0-canary.27.12a4851.0
  ```
</details>

### CODE REVIEW

**Review by kasperpeulen — APPROVED** [review]
Nice!

**Comment by sebaguozden** [comment]
@kasperpeulen @yannbf I tested it with `pnpm create vite` (React -> TypeScript + SWC) and I´m not able to get the coverage correctly (#28)

File: src/webpack5-exclude.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const nycConfig = await getNycConfig({
+    cwd,
+    nycrcPath,
+  });
+
+  return new TestExclude({
+    cwd,
+    include: include ?? nycConfig.include,
+    exclude: exclude ?? nycConfig.exclude ?? defaultExclude,
+    extension: extension ?? nycConfig.extension ?? defaultExtensions,
```
Comment (yannbf) [inline src/webpack5-exclude.ts:21]: does this mean defaultExclude and defaultExtensions won't get merged? So if you decide to exclude `Button.*` you also lose the default exclusion of all sort of files?

Comment (valentinpalkovic) [reply src/webpack5-exclude.ts:21]: And keep in mind, that test-exclude applies some defaults internally as well: - 
- https://www.npmjs.com/package/test-exclude
- https://github.com/istanbuljs/schema/blob/master/default-exclude.js


The `defaultExclude` therefore might even not be necessary?!

**Comment by taozhou-glean** [comment]
how do I expect to run the coverage on the webpack5 example folder ?

I am getting following errors:

```

Test Suites: 3 failed, 3 total
Tests:       8 failed, 8 total
Snapshots:   0 total
Time:        1.416 s
Ran all test suites.
Coverage file (2 bytes) written to .nyc_output/coverage.json
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |
----------|---------|----------|---------|---------|-------------------
```

while `yarn test-storybook` works just fine

```
 PASS   browser: chromium  stories/Page.stories.tsx
 PASS   browser: chromium  stories/Button.stories.tsx
 PASS   browser: chromium  stories/Header.stories.tsx

Test Suites: 3 passed, 3 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        1.318 s
```

**Pull request merged**